src/larktools/evaluation.py

```diff
@@ -85,9 +85,8 @@ def eval_atom(node, env):
     child_name = get_name(child)
     if child_name == "INT":
         return int(get_value(child))
-    elif child_name == "VARNAME":
-        varname = get_value(child)
-        return env[varname]
+    elif child_name == "variable":
```

Comment:
I think your suggested approach is the way to go, not only for `eval_atom` but all node handler functions. Dispatching to node handler functions based on a dictionary with keys given by node names is a more robust and maintainable approach, especially when the evaluation logic becomes more complex. However, I think this would be a good pull request in its own right, so I would suggest to implement this later, and then for all nodes at once.

Reply:
noted in #2